### PR TITLE
Add form

### PR DESCRIPTION
**Pull request by yashodhicy**
🌺 Create a method called current_user in your ApplicationController to make current user data available to all controllers.

1. It will return the first user from the database.

🌺 Create forms to perform the following functions:

1. Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
2. Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

🌺 Allow Users to add likes to Posts.

image

image

image

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
+  
   resources :users, only: [:index,:show] do
-    resources :posts, only: [:index,:show ]
+    resources :posts, only: [:index,:new,:create,:show ]
+  end
+
+  resources :posts do
+    resources :comments, only: [:create]
+    resources :likes, only: [:create]
   end
```
Comment (iambenkis) [inline config/routes.rb]: - [ ] Kindly ensure that there is a pair of routes in the `config/routes.rb` file which corresponds with the `new` and `create`, actions in the comments controller. At this time you have only `create` Kindly add `new` 😄 
Viewing these routes in the command line should be similar to:
```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
   GET   /posts/:post_id/comments/new(.:format)     comments#new
```

- [ ] Simiarly, comments action (`new` &`create`) should be defined in _comments_controller_, to enforce separation of concerns principle. This should be similar to:

```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

 # The new method should be added 
  def new
    @comment = Comment.new
  end

 def create
    @comment = @current_user.comments.new(params)
  	...
 end 
end
```

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+    <%= link_to "Add New Post", new_user_post_path(@user.id) %>
+    <%= link_to "Home", root_path %>
+    <%= link_to "Back", :back %>
+</nav>
+
 <%= render @user%>
+
 <div id="bio">
  Bio: <%= @user.bio %>
  </div>
```
Comment (iambenkis) [inline app/views/users/show.html.erb:14]: - [x] [OPTIONAL] As for the recent posts displayed , users should be able to be redirected whenever they click on them. You did not add that feature on the recent posts , Kindy add them 👍 

```ruby
# In the /views/users/show.html.erb
  <a href ="<%= user_post_path(@user, post)%>">
  <%= render post %>
  </a >
```

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi Team @yashodhicy  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ Nice job using strong params in the posts_controller
✔️ Work documented professionally

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
+  
   resources :users, only: [:index,:show] do
-    resources :posts, only: [:index,:show ]
+    resources :posts, only: [:index,:new,:create,:show ]
+  end
+
+  resources :posts do
+    resources :comments, only: [:new, :create]
```
Comment (mckent05) [inline config/routes.rb:8]: **OPTIONAL** - Good job done so far, but to make your work easier to read and understand, it'd be better having all your routes nested in the first route above. Please fix this 🙏
```ruby
      resources :users, only: [:index, :show] do
          resources :posts, only: [:index, :new, :create, :show] do
               resources :comments only: [:new, :create]
               resources :likes only: [:create]
          end
       end
```

**Review by mckent05 — APPROVED** [review]
Hi @yashodhicy, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job adding forms to your project 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 3 above)
Comment (yashodhicy) [reply config/routes.rb:8]: it says it is not best practice to go more than level 2 deep.

**Pull request merged**